wordcloud/wordcloud.py

```diff
@@ -679,3 +684,45 @@ def __array__(self):
 
     def to_html(self):
         raise NotImplementedError("FIXME!!!")
+
+    def _get_bolean_mask(self, mask):
+        """Cast to two dimensional boolean mask."""
+        if mask.dtype.kind == 'f':
+            warnings.warn("mask image should be unsigned byte between 0"
+                            " and 255. Got a float array")
+        if mask.ndim == 2:
+            boolean_mask = mask == 255
+        elif mask.ndim == 3:
+            # if all channels are white, mask out
+            boolean_mask = np.all(mask[:, :, :3] == 255, axis=-1)
+        else:
+            raise ValueError("Got mask of invalid shape: %s"
+                                % str(mask.shape))
+        return boolean_mask
+
+    def draw_contour(self, img, mask, contour_width=1, contour_color='black'):
+        """Draw mask contour on a pillow image."""
+        mask = self._get_bolean_mask(mask) * 255
+        contour = Image.fromarray(mask.astype(np.uint8))
```

Comment:
Can you not do ``mask[:-1] != mask[1:] + mask[:, :-1] != mask[:, 1:]`` or something like that?

Reply:
I assume you are referring to `_get_boolean_mask`, I didn't write that logic but wanted to reuse it, so put it into a private method `_get_boolean_mask`. I rather not mess with already working code.

Reply:
No, I meant instead of calling "FIND_EDGES" which requires converting to an image and then converting back to a numpy array.